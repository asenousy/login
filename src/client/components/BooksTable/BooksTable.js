import React from 'react'

class BooksTable extends React.Component {
    render() {
        const { maxRows, bookList } = this.props
        const list = bookList.filter(book => !book.volumeInfo.title.startsWith('ISBN'))
            .slice(0, maxRows)
            .map((book, i) => {
                const { volumeInfo: { title, authors, publisher } } = book
                return <tr key={i}>
                    <td>{title}</td>
                    <td>{authors.join(',')}</td>
                    <td>{publisher}</td>
                </tr>
            })

        if (list.length === 0) {
            return null
        }

        return (
            <table className='table'>
                <thead>
                    <tr>
                        <th scope="col">Book Name</th>
                        <th scope="col">Author Name</th>
                        <th scope="col">Publisher Name</th>
                    </tr>
                </thead>
                <tbody>
                    {list}
                </tbody>
            </table>
        )
    }
}

export default BooksTable
