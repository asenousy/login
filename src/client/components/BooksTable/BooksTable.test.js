import React from 'react';
import BooksTable from './BooksTable';
import renderer from 'react-test-renderer';

it('renders table correctly', () => {
    const books = [{
        volumeInfo: {
            title: 'title',
            authors: ['author'],
            publisher: 'publisher'
        }
    }]

    const tree = renderer
        .create(<BooksTable maxRows={1} bookList={books} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})
