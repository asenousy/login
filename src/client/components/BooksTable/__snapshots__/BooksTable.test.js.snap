// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Book Name
      </th>
      <th
        scope="col"
      >
        Author Name
      </th>
      <th
        scope="col"
      >
        Publisher Name
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        title
      </td>
      <td>
        author
      </td>
      <td>
        publisher
      </td>
    </tr>
  </tbody>
</table>
`;
